python/lsst/daf/butler/core/butlerConfig.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Configuration classes specific to the Butler
+"""
+
+import os
+
+from .config import Config
+from .utils import doImport
+
+__all__ = ("ButlerConfig",)
+
+
+class ButlerConfig(Config):
+    """Contains the configuration for a `Butler`
+
+    The configuration is read and merged with default configurations for
+    the particular classes. The defaults are read from
+    ``$DAF_BUTLER_DIR/config`` using names specified by the appropriate classes
+    for registry and datastore, and including the standard schema and storage
+    class definitions.
+
+    Parameters
+    ----------
+    other : `str`
+        Path to butler configuration YAML file.
+    """
+
+    def __init__(self, other):
+
+        # Create an empty config for us to populate
+        super().__init__()
+
+        # Find the butler configs
+        self.defaultsDir = None
+        if "DAF_BUTLER_DIR" in os.environ:
+            self.defaultsDir = os.path.join(os.environ["DAF_BUTLER_DIR"], "config")
```

Comment:
`utils.getPackageDir`?

Reply:
I wondered about that. `utils` would work fine but I do wonder about adding LSST dependencies for no real gain. We only use `utils` in tests and I would argue that we don't need that dependency. That leaves only `lsst.log` and `sphgeom` stopping us from putting this package directly on pypi for everyone to use without needing any non-standard dependencies. (I'm thinking about world domination here...)

Reply:
I have some of the same thoughts about world domination (see RFC-430 and RFC-460 vis a vis getting that sphgeom dependency out on pypi).  But I think using `DAF_BUTLER_DIR` here just avoids an explicit dependency on `utils` via an implicit dependency on `EUPS` and perhaps `sconsUtils`, and I think the `utils` version may be slightly easier to rip out and replace with something unencumbered later.